Fix HTTP::request_headers

WinterSilence opened the pull request:
Trim 'HTTP_' only at the beginning, converts to lowercase

WinterSilence committed:
Fix HTTP::request_headers

Trim 'HTTP_' only at the beginning, converts to lowercase

cs278 commented on classes/Kohana/HTTP.php:
```diff
@@ -180,14 +180,13 @@ public static function request_headers()
 			{
-				continue;
+				// This is a dirty hack to ensure HTTP_X_FOO_BAR becomes x-foo-bar
+				$key = str_replace('_', '-', strtolower(ltrim($key, 'HTTP_')));
```
`HTTP_HOST` becomes `ost`, use `substr()`.

WinterSilence committed:
Update HTTP.php

enov commented:
Excellent @WinterSilence. Should probably go to `3.3/develop` as it is a non-breaking bug fix.

WinterSilence committed:
Update HTTP.php

https://github.com/kohana/core/blob/3.4/develop/classes/Kohana/HTTP/Header.php#L297

WinterSilence committed:
Fix HTTP::request_headers

Remove strtolower for keys, see https://github.com/kohana/core/blob/3.4/develop/classes/Kohana/HTTP/Header.php#L297
Added support for HTTP extension version 2

WinterSilence commented:
@enov https://github.com/kohana/core/pull/617 also add some other fixes
